services/cargohold/src/CargoHold/S3.hs

```diff
@@ -706,147 +721,46 @@ getAmzMetaChunkSize :: [(Text, Text)] -> Maybe V3.ChunkSize
 getAmzMetaChunkSize = parseAmzMeta hAmzMetaChunkSize
 
 getAmzMetaUploadId :: [(Text, Text)] -> Maybe Text
-getAmzMetaUploadId = lookup hAmzMetaUploadId
+getAmzMetaUploadId = lookupCI hAmzMetaUploadId
 
 parseAmzMeta :: FromByteString a => Text -> [(Text, Text)] -> Maybe a
-parseAmzMeta k h = lookup k h >>= fromByteString . encodeUtf8
+parseAmzMeta k h = lookupCI k h >>= fromByteString . encodeUtf8
 
 -------------------------------------------------------------------------------
 -- Utilities
 
-tryS3 :: ResourceT App a -> ExceptT Error App a
-tryS3 action = do
-  e <- lift ask
-  runAppResourceT e action
-    `catch` \(ex :: S3Error) -> case s3ErrorCode ex of
-      "InvalidDigest" -> throwE invalidMD5
-      "BadDigest" -> throwE invalidMD5
-      "RequestTimeout" -> throwE requestTimeout
-      _ -> do
-        lift . Log.err $ msg (show ex)
-        throwE serverError
-
-exec ::
-  (Aws.Transaction r a, ServiceConfiguration r ~ S3Configuration) =>
-  r ->
-  ResourceT App (ResponseMetadata a, a)
-exec req = do
-  env <- awsEnv <$> view aws
-  s3c <- s3Config <$> view aws
-  Aws.sendRequest env s3c req
-
-x3 :: RetryPolicy
-x3 = limitRetries 3 <> exponentialBackoff 200000
-
-handlers :: Monad m => [RetryStatus -> Handler m Bool]
-handlers =
-  Retry.httpHandlers
-    ++ [ const $ Handler $ \(S3Error s _ _ _ _ _ _ _ _ _) ->
-           return $ statusIsServerError s
-       ]
+parseMIMEType :: Text -> Maybe MIME.Type
+parseMIMEType = MIME.parseMIMEType
```

Comment:
Is it worth having this as it's just an alias?

Reply:
No, will inline it. Same for `encodeMIMEType`.

Reply:
https://github.com/wireapp/wire-server/pull/1157/commits/877c9a96036ec475d1506ad738621b2ffe417cc6